test/aion-abi.encoding.js

```diff
@@ -0,0 +1,163 @@
+var chai = require('chai');
+var assert = chai.assert;
+var Abi = require('../packages/aion-lib/src/abi.js');
+
+var encodeEventSignatureTests = [{
+    params: [{
+        name: 'myEvent',
+        type: 'event',
+        inputs: [{
+            type: 'uint128',
+            name: 'myNumber'
+        },{
+            type: 'bytes32',
+            name: 'myBytes'
+        }]
+    }],
+    result: '0x2236226bdca8de2f2bc9a5be931cb54acc7d73c5a829b22dc2b7e92af174f7d4'
+},{
+    params: [{
+        name: 'SomeEvent',
+        type: 'event',
+        inputs: [{
+            type: 'bytes',
+            name: 'somebytes'
+        },{
+            type: 'byte16',
+            name: 'myBytes'
+        }]
+    }],
+    result: '0xab132b6cdd50f8d4d2ea33c3f140a9b3cf40f451540c69765c4842508bb13838'
+},{
+    params: [{
+        name: 'AnotherEvent',
+        type: 'event',
+        inputs: []
+    }],
+    result: '0x601d819e31a3cd164f83f7a7cf9cb5042ab1acff87b773c68f63d059c0af2dc0'
+}];
+
+var encodeFunctionSignatureTests = [{
+	params:[{
+		name: 'myFunction',
+		type: 'function',
+		inputs: [{
+			type: 'uint128',
+			name: 'myNumber'
+		},{
+			type: 'bytes32',
+			name: 'myBytes'
+		}]
+	}],
+	result: '0xeabf647b8f8eda8815ce2b4b17c04ffa8a90128864990c859e7a3f65282e9802'.slice(0, 6)
+},{
+	params:[{
+		name: 'myFunction',
+		type: 'function',
+		inputs: [{
+			type: 'uint128',
+			name: 'myNumber'
+		},{
+			type: 'bytes32',
+			name: 'myBytes'
+		}]
+	}],
+	result: '0xeabf647b8f8eda8815ce2b4b17c04ffa8a90128864990c859e7a3f65282e9802'.slice(0, 6)
```

Comment:
Why is slice(0,6) needed?

Reply:
in the source code, after hashing the values, it has the following line that slices and only takes in the first 6 index.
".slice(0, values.solidity.types.function.byteLengthEncoded)"